rbs/MethodTypeToParserNode.cc

```diff
@@ -69,6 +81,69 @@ core::NameRef nodeName(const parser::Node *node) {
     return name;
 }
 
+string argKindToString(RBSArg::Kind kind) {
+    switch (kind) {
+        case RBSArg::Kind::Positional:
+            return "positional";
+        case RBSArg::Kind::OptionalPositional:
+            return "optional positional";
+        case RBSArg::Kind::RestPositional:
+            return "rest positional";
+        case RBSArg::Kind::Keyword:
+            return "keyword";
+        case RBSArg::Kind::OptionalKeyword:
+            return "optional keyword";
+        case RBSArg::Kind::RestKeyword:
+            return "rest keyword";
+        case RBSArg::Kind::Block:
+            return "block";
+    }
+}
+
+string nodeKindToString(const parser::Node *node) {
+    string kind;
+
+    typecase(
+        node, [&](const parser::Arg *parserArg) { kind = "positional"; },
+        [&](const parser::Optarg *parserArg) { kind = "optional positional"; },
+        [&](const parser::Restarg *parserArg) { kind = "rest positional"; },
+        [&](const parser::Kwarg *parserArg) { kind = "keyword"; },
+        [&](const parser::Kwoptarg *parserArg) { kind = "optional keyword"; },
+        [&](const parser::Kwrestarg *parserArg) { kind = "rest keyword"; },
+        [&](const parser::Blockarg *parserArg) { kind = "block"; },
+        [&](const parser::Node *other) {
+            Exception::raise("Unexpected expression type: {}", ((parser::Node *)node)->nodeName());
+        });
+
+    return kind;
+}
+
+bool checkParameterKind(core::MutableContext ctx, const RBSDeclaration &declaration, const RBSArg &arg,
+                        const parser::Node *methodArg) {
+    auto error = false;
```

Comment:
Can we rename this to something like `kindMatches`? When I saw `error`, I expected it hold some sort of error object, not a boolean.